osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -187,17 +210,19 @@ public async Task ExpirePlaylistItemAsync(long playlistItemId)
         public async Task EndMatchAsync(MultiplayerRoom room)
         {
             // Remove all non-expired items from the playlist as they have no scores.
-            await connection.ExecuteAsync(
-                "DELETE FROM multiplayer_playlist_items p WHERE p.room_id = @RoomID AND p.expired = 0 AND (SELECT COUNT(*) FROM multiplayer_scores s WHERE s.playlist_item_id = p.id) = 0",
+            await connection.ExecuteAsync("DELETE FROM multiplayer_playlist_items p WHERE p.room_id = @RoomID AND p.expired = 0 AND (SELECT COUNT(*) FROM multiplayer_scores s WHERE s.playlist_item_id = p.id) = 0",
                 new
                 {
                     RoomID = room.RoomID
                 });
 
+            int totalUsers = connection.QuerySingle<int>("SELECT COUNT(*) FROM multiplayer_rooms_high WHERE room_id = @RoomID", new { RoomID = room.RoomID });
```

Comment:
I'm confused, what exactly is this doing? Setting `participant_count` to 1?

Reply:
to the total count of users (that ever participated in the room).

Reply:
Ah, I see. I thought it was selecting `multiplayer_rooms`, not `multiplayer_rooms_high`.